src/experimental-layers/src/mesh-layer/mesh-layer-fragment.glsl.js

```diff
@@ -16,17 +16,13 @@ varying float vLightWeight;
 void main(void) {
   // TODO - restore color rendering
 
-  gl_FragColor = vColor / 255.;
-
-  // hasTexture ? texture2D(sampler, vTexCoord) : color / 255.;
-  // color = vec4(color_transform(color.rgb), color.a * opacity);
-
-  // gl_FragColor = gl_FragColor * vLightWeight;
+  vec4 color = hasTexture ? texture2D(sampler, vTexCoord) : vColor / 255.;
+  gl_FragColor = vec4(color.rgb * vLightWeight, color.a);
 
   // use highlight color if this fragment belongs to the selected object.
-  // gl_FragColor = picking_filterHighlightColor(gl_FragColor);
+  gl_FragColor = picking_filterHighlightColor(gl_FragColor);
```

Comment:
Is there a combined `picking_filterColor` that calls both of these for the standard case when no intermediary processing is needed?

Reply:
@1chandu

Reply:
No, we don't have one, but I see the pattern we could add one.